android/app/src/main/java/app/organicmaps/location/LocationHelper.java

```diff
@@ -169,6 +177,20 @@ private void notifyLocationUpdated()
         mSavedLocation.getBearing());
   }
 
+  private void notifyLocationUpdateTimeout()
+  {
+    mHandler.removeCallbacks(mLocationTimeoutRunnable);
```

Comment:
This removal will arise a case in which runnable will be removed from empty handler. 

steps 
1 - on start runnable will be added in handler
2 - if location update stops it will trigger notifyLocationUpdateTimeout() which will remove runnable.
3 - Now again location gets fixed and it updates then runnable will be removed which do not exists in handler.